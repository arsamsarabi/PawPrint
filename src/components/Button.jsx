import React from "react";
import { Text, StyleSheet, TouchableOpacity } from "react-native";
import Icon from "react-native-vector-icons/FontAwesome";

export const Button = ({ text, icon, onClick, mode }) => {
  const styles = (el) => {
    switch (mode) {
      case "Large":
        return largeBtnStyles[el];
      case "Icon":
        return iconBtnStyles[el];
      default:
        return "";
    }
  };
  return (
    <TouchableOpacity onPress={onClick} style={styles("actionBtn")}>
      <Icon name={icon} size={24} color="#846d63" />
      <Text style={styles("btnText")}>{text}</Text>
    </TouchableOpacity>
  );
};

const iconBtnStyles = StyleSheet.create({
  actionBtn: {
    alignItems: "center",
    color: "#846d63",
    flexDirection: "row",
    justifyContent: "center",
    marginLeft: 25
  },
  btnText: {
    color: "#846d63",
    fontSize: 16,
    marginLeft: 5
  }
});

const largeBtnStyles = StyleSheet.create({
  actionBtn: {
    alignItems: "center",
    alignSelf: "stretch",
    backgroundColor: "#cbc4b8",
    flexDirection: "row",
    height: 60,
    justifyContent: "center"
  },
  btnText: {
    color: "#846d63",
    fontSize: 24,
    marginLeft: 20
  }
});
