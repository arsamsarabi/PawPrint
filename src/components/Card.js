import React from "react";
import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  ImageBackground
} from "react-native";

export const Card = ({ name, age, breed, image }) => {
  const placeholderImage = require("../../assets/placeholder.jpg");
  return (
    <View style={styles.container}>
      <ImageBackground
        source={image || placeholderImage}
        style={styles.imageBg}
      >
        <View style={styles.footer}>
          <View style={styles.detail}>
            <Text style={styles.text}>{name || "Name: -"}</Text>
          </View>
          <View style={[styles.detail, styles.borders]}>
            <Text style={styles.text}>{`Age: ${age || "-"}`}</Text>
          </View>
          <View style={styles.detail}>
            <Text style={styles.text}>{breed || "Breed: -"}</Text>
          </View>
        </View>
      </ImageBackground>
    </View>
  );
};

const CONTAINER_WIDTH = Dimensions.get("window").width - 60;

const styles = StyleSheet.create({
  container: {
    width: CONTAINER_WIDTH,
    overflow: "hidden",
    borderRadius: 10,
    marginBottom: 30
  },
  imageBg: {
    width: CONTAINER_WIDTH,
    height: CONTAINER_WIDTH,
    justifyContent: "flex-end"
  },
  footer: {
    flexDirection: "row",
    backgroundColor: "#846d63c0",
    height: 50
  },
  detail: {
    flex: 1,
    padding: 5,
    borderRadius: 10,
    justifyContent: "center",
    alignItems: "center"
  },
  borders: {
    borderColor: "#cbc4b8",
    borderRightWidth: 1,
    borderLeftWidth: 1,
    borderRadius: 0
  },
  text: {
    fontSize: 16,
    fontWeight: "bold",
    color: "#ededed"
  }
});
