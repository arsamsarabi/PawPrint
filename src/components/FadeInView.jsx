import React, { useState, useEffect } from "react";
import { Animated } from "react-native";

export const FadeInView = ({ style, children }) => {
  const [fadeAnim] = useState(new Animated.Value(0));

  useEffect(() => {
    Animated.timing(fadeAnim, {
      toValue: 1,
      duration: 1000
    }).start();
  }, []);

  return (
    <Animated.View
      style={{
        ...style,
        opacity: fadeAnim
      }}
    >
      {children}
    </Animated.View>
  );
};
