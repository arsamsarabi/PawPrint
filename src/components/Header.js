import React from "react";
import { Text, Image, StyleSheet, View } from "react-native";

export const Header = () => {
  return (
    <View style={styles.header}>
      <Image style={styles.logo} source={require("../../assets/logo.png")} />
      <Text style={styles.title}>Paw Print</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    alignItems: "center",
    paddingTop: 10,
    marginBottom: 15
  },
  logo: {
    height: 75,
    width: 75
  },
  title: {
    fontSize: 22,
    marginTop: 10,
    color: "#846d63",
    fontWeight: "bold"
  }
});
