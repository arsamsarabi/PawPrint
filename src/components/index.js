export * from "./Header";
export * from "./Card";
export * from "./Button";
export * from "./FadeInView";
