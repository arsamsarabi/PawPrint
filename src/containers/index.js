export * from "./List";
export * from "./LandingPage";
export * from "./Form";
export * from "./DogModal";
