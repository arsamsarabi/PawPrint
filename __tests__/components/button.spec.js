import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { Button } from "../../src/components";

describe("<Button />", () => {
  test("Should render a large button", () => {
    const component = renderer
      .create(
        <Button
          text="Large Button"
          icon="times"
          onClick={() => jest.fn()}
          mode="Large"
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
