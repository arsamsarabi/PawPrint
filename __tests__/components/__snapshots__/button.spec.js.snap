// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render a large button 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#cbc4b8",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  times
  <Text
    style={
      Object {
        "color": "#846d63",
        "fontSize": 24,
        "marginLeft": 20,
      }
    }
  >
    Large Button
  </Text>
</View>
`;
